go/adbc/driver/snowflake/snowflake_database.go

```diff
@@ -328,13 +330,28 @@ func (d *databaseImpl) SetOptions(cnOptions map[string]string) error {
 				}
 			}
 
-			d.cfg.PrivateKey, err = x509.ParsePKCS1PrivateKey(data)
+			var block []byte
+			if strings.Contains(string(data), "PRIVATE KEY") {
+				b, _ := pem.Decode(data)
+				block = b.Bytes
+			} else {
+				block = data
+			}
+
+			var key any
+			key, err = x509.ParsePKCS1PrivateKey(block)
+			if err != nil && strings.Contains(err.Error(), "use ParsePKCS8PrivateKey instead") {
+				key, err = x509.ParsePKCS8PrivateKey(block)
+			}
+			
 			if err != nil {
 				return adbc.Error{
 					Msg:  "failed parsing private key file '" + v + "': " + err.Error(),
 					Code: adbc.StatusInvalidArgument,
 				}
 			}
+
+			d.cfg.PrivateKey = key
```

Comment:
You can't make this assign directly, you need to check that `key` is an `*rsa.PrivateKey` and type assert it. 

While `ParsePKCS1PrivateKey` returns an `*rsa.PrivateKey`, `ParsePKCS8PrivateKey` might return an `*ecdsa.PrivateKey`, an `ed25519.PrivateKey` (not a pointer), or an `*ecdh.PrivateKey` (for X25519) while the Snowflake driver currently only accepts an `*rsa.PrivateKey` as far as I'm aware.

Reply:
see https://github.com/snowflakedb/gosnowflake/blob/dfb1c18624ebd4023e27839cd32a51955758bd75/priv_key_test.go#L52 for an example